api/types/userloginstate/user_login_state.go

```diff
@@ -53,6 +53,12 @@ type Spec struct {
 	// Traits are the traits attached to the user login state.
 	Traits trait.Traits `json:"traits" yaml:"traits"`
 
+	// InheritedRoles are any Roles that were inherited via inclusion in nested Access Lists.
+	InheritedRoles []string `json:"inherited_roles" yaml:"inherited_roles"`
+
+	// InheritedTraits are any Traits that were inherited via inclusion in nested Access Lists.
+	InheritedTraits trait.Traits `json:"inherited_traits" yaml:"inherited_traits"`
```

Comment:
This object is stored in the backend side and right now it looks like these two properties were added only to emit usage audit event. 

I think that using a object that is stored on backend side just to forward few arguments from nested function is bad practice. 

The backend object should represent the Domain but right now the `InheritedRoles`, InheritedTraits where  we are adding a some garbage metadata to the UserLogicState. 

I would suggest collect and forward InheritedRoles and InheritedTraits directly in the code without using backend storage object.